### PR TITLE
15950-Set out of stock status when import product though CSV

### PR DESCRIPTION
**Pull request by saravananvelu**
### Description
This pull request is created to set the out of stock status with the quantity greater than zero when importing the products through CSV.

### Fixed Issues
* magento/magento2 #15950 : Magento2 CSV product import qty and is_in_stock not working correct

### Manual testing scenarios
Importing the simple products through CSV with the quantity greater than zero and is_in_stock field value is zero.
	
Path to import products:
1. Choose System -> Import under Data transfer
1. Choose Products from the Entity dropdown
1. Choose Add/Update from the Import Behavior dropdown
1. Keep other fields as default
1. Import the products
1. Now check the product status is out of stock with the quantity as given in CSV.

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @saravananvelu. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on Pull Request changes
- `@magento-engcom-team give me new test instance` - deploy **NEW** test instance based on Pull Request changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy Vanilla Magento instance for Issue or Pull Request

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by saravananvelu** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @saravananvelu. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @saravananvelu, here is your [Magento instance](https://pr-16084.engcom.dev.magento.com).
Admin access: [https://pr-16084.engcom.dev.magento.com/admin](https://pr-16084.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Please make sure you are PR **author** or **assignee** to access the instance.

**Comment by saravananvelu** [comment]
@magento-engcom-team  Pull request working as expected in the provided test environment

File: app/code/Magento/CatalogInventory/Model/StockStateProvider.php [hunk 1]
```diff
@@ -80,6 +80,9 @@ public function verifyStock(StockItemInterface $stockItem)
         ) {
             return false;
         }
+        if ($stockItem->getQty() > 0 && !$stockItem->getIsInStock()) {
+            return false;
```
Comment (orlangur) [inline app/code/Magento/CatalogInventory/Model/StockStateProvider.php:84]: Why this logic needed in `CatalogInventory` if only `ImportExport` is affected?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/CatalogInventory/Model/StockStateProvider.php (hunk 1 above)
Comment (saravananvelu) [reply app/code/Magento/CatalogInventory/Model/StockStateProvider.php:84]: That specific function will have a levels of the stock check while product import/export and which does not cause any other problem for other behaviors

**Review by saravananvelu — COMMENTED** [review]
I keep the same code.

**Comment by orlangur** [comment]
Waiting for decision on alternative implementation by @p-bystritsky delivery.

**Comment by orlangur** [comment]
Still not clear about the delivery plan for https://github.com/magento-engcom/magento2ce/pull/2146.

**Comment by sidolov** [comment]
Hi @saravananvelu looks like this issue was fixed in scope of https://github.com/magento/magento2/pull/20177
I'm closing this PR as not relevant. 
Thank you for collaboration!

**Pull request closed without merging**